include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -98,7 +90,35 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 	public:
 		virtual ~CSummedAreaTableImageFilter() {}
 
-		class CStateBase : public CMatchedSizeInOutImageFilterCommon::state_type, public CSummedAreaTableImageFilterBase<ExclusiveMode>::CSummStateBase { friend class CSummedAreaTableImageFilter<ExclusiveMode>; };
+		class CStateBase : public CMatchedSizeInOutImageFilterCommon::state_type, public CSummedAreaTableImageFilterBase<ExclusiveMode>::CSummStateBase 
+		{ 
+			public:
+				CStateBase() = default;
+				virtual ~CStateBase() = default;
+
+			private:
+
+				CStateBase(const CStateBase& referenceState)
+				{
+					this->inImage = referenceState.inImage; 
+					this->outImage = referenceState.outImage;
+					this->inOffset = referenceState.inOffset;
+					this->inBaseLayer = referenceState.inBaseLayer;
+					this->outOffset = referenceState.outOffset;
+					this->outBaseLayer = referenceState.outBaseLayer;
+					this->scratchMemoryByteSize = referenceState.scratchMemoryByteSize;
+					this->extent = referenceState.extent;
+					this->layerCount = referenceState.layerCount;
+					this->inMipLevel = referenceState.inMipLevel;
+					this->outMipLevel = referenceState.outMipLevel;
+					this->normalizeImageByTotalSATValues = referenceState.normalizeImageByTotalSATValues;
+
+					this->scratchMemory = reinterpret_cast<uint8_t*>(_IRR_ALIGNED_MALLOC(referenceState.scratchMemoryByteSize, 8));
+					memcpy(this->scratchMemory, referenceState.scratchMemory, this->scratchMemoryByteSize);
```

Comment:
do not reallocate memory and copy scratch

scratch == memory to be used temporarily and contents to be thrown away (not preserved)

Reply:
corrected in following commit